landing-page/scripts/staffline_stage.py

```
@@ -180,10 +188,140 @@ def _assemble_jsomr_records(image_name, fit_results, boxes, grouping_result, sca
+        accepted, cap_exceeded = validate_and_select_candidates(region, candidates, sibling_widths)
+        trace.append(
+            f"[trace] {image_name}: stave {region.stave_id} fallback re-probe: "
+            f"{len(candidates)} candidate(s) found, {len(accepted)} accepted"
+            + (f", cap_exceeded (fallback_found_more_than_expected:{region.stave_id})" if cap_exceeded else "")
+        )
+
+        for candidate in accepted:
+            fit_result = candidate.fit
+            fit_result.flags = list(fit_result.flags) + [
+                "fallback_redetected", f"fallback_conf:{candidate.yolo_confidence:.3f}",
+            ]
```

Ruff (0.16.1)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)